docs/versions/latest/Components/Archetypes/typescript/Components/Policy.md

```diff
@@ -0,0 +1,208 @@
+A Example typescript policy with all posible configurations.
+
+~~~ts
+import { PolicyTemplate } from "vrotsc-annotations";
+
+@PolicyTemplate({
+    name: "Sample Policy",
+    path: "MyOrg/MyProject",
+    description: "Some Description",
+    variables: {
+        sample: {
+            type: "string",
+            value: "a string value with type mentioned",
+            description: "A variable created with type and description"
+        },
+        config: {
+            type: "Properties",
+            configId: "8e2d3ba0-4e2c-4d4c-ad82-76de4967bf9f",
+            configKey: "props",
+            description: "A variable created with configuration binding"
+        },
+        sample2: "a direct string value"
+    },
+    elements: {
+        ElementOne: {
+            type: "AMQP:Subscription",
+            events: {
+                onMessage: "onMessage"
+            }
+        },
+        ElementTwo: {
+            type: "SNMP:SnmpDevice",
+            events: {
+                OnTrap: {
+                    workflowId: "56652d1b-f797-3e1c-a108-7db9ddfb1f42",
+                    bindings: {
+                        workflowInputOne: {
+                            type: "string",
+                            variable: "sample"
+                        }
+                    }
+                }
+            }
+        }
+        ElementThree: {
+            type: "Periodic Event",
+            schedule: {
+                periode: "every-minutes",
+                when: "13,15",
+                timezone: "Asia/Calcutta"
+            }
+        }
+    }
+})
+~~~
+
+## Creating a Variable in Policy
+Variables can be created in a policy by adding them under **variables** attribute as shown in the example.
+
+### Creating a String variable
+String varibales can addes by directly assigning values to the them.
+~~~ts
+variables: {
+    sample2: "a direct string value"
+}
+~~~
+
+### Creating Variables with other data types
+Other data type variables can be added by providing the additional information to them.
+~~~ts
+variables: {
+    sample: {
+        type: "boolean",
+        value: true,
+        description: "A variable created with type and description"
+    },
+    sample2:{
+        type: "number",
+        value: 2,
+        description: "A Numeric value"
+    }
+}
+~~~
+
+### Creating Variables with Configuration binding
+Variables can be binded to configurations by providing configuration ID and key.
+
+~~~ts
+variables: {
+    config: {
+        type: "Properties",
+        configId: "8e2d3ba0-4e2c-4d4c-ad82-76de4967bf9f",
+        configKey: "props",
+        description: "A variable created with configuration binding"
+    }
+}
+~~~
+
+**Note:** Configuration binding will be considered if both value and configuration ID are provided to a variable.
+
+## Creating a Elements to a Policy
+Elements can be created in a policy by adding them under **elements** attribute as shown in example
+
+### Supported Element types
```

Comment:
It's not a limitation, it unlocks the full capability support of the Orchestrator Policy by ABT

Reply:
I fail to understand how this unlocks the full capabilities?
We have a plugin for Kafka, that plugin defines a custom Kafka Consumer element, how would this allow us to use that plugin? Looking at the code, it seems like it should be fine as long as we modify:

```ts
type VroPolicyTemplateType = "AMQP:Subscription" | "MQTT:Subscription" | "SNMP:SnmpDevice" | "SNMP:TrapHost" | "Periodic Event";
```

to be something like:

```ts
type VroPolicyTemplateType = strung | "AMQP:Subscription" | "MQTT:Subscription" | "SNMP:SnmpDevice" | "SNMP:TrapHost" | "Periodic Event";

```

Reply:
Instead of adding string to the type, We can directly assign string alone as its types, as the adding of string type will no longer generate the list of elements in intellisense. Please suggest.

Reply:
We can also try:
~~~ts
type VroPolicyTemplateType = (string & {}) | "AMQP:Subscription" | "MQTT:Subscription" | "SNMP:SnmpDevice" | "SNMP:TrapHost";
~~~
this can still show the list of elements types and still accept a string.

Reply:
You will get intellisense if you add string, no problem. You won't get that it's an error in case you specify something out of the defaults, but that is alright

Reply:
Added string type to policy types.